python/cog/server/clients.py

```diff
@@ -0,0 +1,319 @@
+import base64
+import io
+import mimetypes
+import os
+from typing import (
+    Any,
+    AsyncIterator,
+    Awaitable,
+    Callable,
+    Collection,
+    Dict,
+    Mapping,
+    Optional,
+    cast,
+)
+from urllib.parse import urlparse
+
+import httpx
+import structlog
+
+from .. import types
+from ..schema import Status, WebhookEvent
+from ..types import Path
+from .eventtypes import PredictionInput
+from .response_throttler import ResponseThrottler
+from .retry_transport import RetryTransport
+from .telemetry import current_trace_context
+
+log = structlog.get_logger(__name__)
+
+
+def _get_version() -> str:
+    try:
+        try:
+            from importlib.metadata import version
+        except ImportError:
+            pass
+        else:
+            return version("cog")
+        import pkg_resources
+
+        return pkg_resources.get_distribution("cog").version
+    except Exception:
+        return "unknown"
+
+
+_user_agent = f"cog-worker/{_get_version()} {httpx._client.USER_AGENT}"
+_response_interval = float(os.environ.get("COG_THROTTLE_RESPONSE_INTERVAL", 0.5))
+
+# HACK: signal that we should skip the start webhook when the response interval
+# is tuned below 100ms. This should help us get output sooner for models that
+# are latency sensitive.
+SKIP_START_EVENT = _response_interval < 0.1
+
+WebhookSenderType = Callable[[Any, WebhookEvent], Awaitable[None]]
+
+
+def common_headers() -> "dict[str, str]":
+    headers = {"user-agent": _user_agent}
+    return headers
+
+
+def webhook_headers() -> "dict[str, str]":
+    headers = common_headers()
+    auth_token = os.environ.get("WEBHOOK_AUTH_TOKEN")
```

Comment:
I don't think this is needed any more.

Reply:
it seems like it's still in main? https://github.com/replicate/cog/blob/main/python/cog/server/webhook.py#L73